pandas/core/generic.py

```diff
@@ -3272,6 +3272,9 @@ def convert_objects(self, convert_dates=True, convert_numeric=False,
     @Appender(_shared_docs['fillna'] % _shared_doc_kwargs)
     def fillna(self, value=None, method=None, axis=None, inplace=False,
                limit=None, downcast=None):
+        if is_integer(limit) and not limit > 0:
```

Comment:
a) I would like these to be handled in the current cython code. b) Alternatively you can take out the check there and just put an assertion. I don't really like these types of checks being separated from the actual code, though so prefer a)